jaxopt/_src/nonlinear_cg.py

```diff
@@ -180,23 +182,29 @@ def update(self,
                                       grad,
                                       None, # descent_direction
                                       *args, **kwargs)
+      new_value = ls_state.value
+      new_aux = ls_state.aux
+      new_params = ls_state.params
+      new_grad = ls_state.grad
 
     elif self.linesearch == "zoom":
       ls_state = zoom_linesearch(f=self._value_and_grad_with_aux,
                                  xk=params, pk=descent_direction,
                                  old_fval=value, gfk=grad, maxiter=self.maxls,
                                  value_and_grad=True, has_aux=True,
                                  args=args, kwargs=kwargs)
+      new_value = ls_state.f_k
+      new_aux = ls_state.aux
       new_stepsize = ls_state.a_k
+      new_grad = ls_state.g_k
+      # FIXME: zoom_linesearch currently doesn't return new_params
+      # so we have to recompute it.
+      t = new_stepsize.astype(tree_single_dtype(params))
+      new_params = tree_add_scalar_mul(params, t, descent_direction)
 
     else:
       raise ValueError("Invalid name in 'linesearch' option.")
 
-    new_params = tree_add_scalar_mul(params, new_stepsize, descent_direction)
-    (new_value, new_aux), new_grad = self._value_and_grad_with_aux(new_params,
-                                                                   *args,
-                                                                   **kwargs)
```

Comment:
This extra call was only needed to compute values that were already accessible in the outputs of the line search methods, except for `new_aux`.
Now that `new_aux` is available in the output of the linesearch solvers, we can get rid of this call.

Note that this differs from the BFGS and L-BFGS solvers because these quasi-Newton solvers support a stepsize API.